src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+            if (player->transformation == PLAYER_FORM_GORON) {
+                if ((gSaveContext.save.weekEventReg[0x4D] & 0x80)) {
+                    if (this->textID == 0xDF1) {
+                        this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                        return 0xDF2;
+                    }
+                    return 0xDF1;
+                }
+                this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDF0;
+            }
+            switch (this->textID) {
+                case 0xDED:
+                    return 0xDEE;
+                case 0xDEE:
+                    this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                    return 0xDEF;
+                default:
+                    return 0xDED;
+            }
+            break;
+
+        case ENSGORO_TYPE_SHRINE_SPRING_E:
+            if (player->transformation == PLAYER_FORM_GORON) {
+                if ((gSaveContext.save.weekEventReg[0x4D] & 0x80)) {
+                    if (this->textID == 0xDF6) {
+                        this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                        return 0xDF7;
+                    }
+                    return 0xDF6;
+                }
+                this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDF5;
+            }
+            if (this->textID == 0xDF3) {
+                this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                return 0xDF4;
+            }
+            return 0xDF3;
+
+        case ENSGORO_TYPE_SHRINE_SPRING_F:
+            if (player->transformation == PLAYER_FORM_GORON) {
+                if ((gSaveContext.save.weekEventReg[0x4D] & 0x80)) {
+                    switch (this->textID) {
+                        case 0xDFB:
+                            return 0xDFC;
+                        case 0xDFC:
+                            this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                            return 0xDFD;
+                        default:
+                            return 0xDFB;
+                    }
+                } else {
+                    this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                    return 0xDFA;
+                }
+            } else {
+                if (this->textID == 0xDF8) {
+                    this->actionFlags |= ENSGORO_ACTIONFLAG_LASTMESSAGE;
+                    return 0xDF9;
+                }
+                return 0xDF8;
+            }
+            break;
+
+        default:
+            return 0x0;
+    }
+}
+
+/*
+        Dialogue tree for bomb shop goron. Returns ID of next message to display.
+*/
+u16 EnSGoro_BombshopGoron_NextTextID(EnSGoro* this, PlayState* play) {
+
+    Player* player = GET_PLAYER(play);
+
+    switch (this->textID) {
+        case 0x0:
+            switch (player->transformation) {
+
+                case PLAYER_FORM_GORON:
+                    // Powder keg in powder keg slot?
```

Comment:
```suggestion
                    // Powder keg in powder keg slot
```
No need for `?` that is what is indeed checking